lib/hanami/cli/commands/gem/new.rb

```diff
@@ -36,12 +39,15 @@ def initialize(
           end
           # rubocop:enable Metrics/ParameterLists
 
-          def call(app:, skip_bundle: SKIP_BUNDLE_DEFAULT, **)
-            app = inflector.underscore(app)
+          def call(app_path:, skip_bundle: SKIP_BUNDLE_DEFAULT, **)
+            raise ArgumentError, <<~MSG if app_path == ROOT_PATH
+              System's root directory is not allowed as the application path
+            MSG
```

Comment:
The way we phrase this special case seems undesirable to me.

Like, if I am root on a machine and I _want_ my hanami app to live at `/`, why _shouldn't_ I be able to put it there?

Now, I think the actual problem here is that we can't determine an app name if. the given `app_path` has no actual word as its basename.

This IMO is actually another reason to add `--app-name` as an option. Once we do that, we can move the location of this error to be the part of our code that determines a default app name from the given `app_path` if there is no explicit `--app-name` option provided. From there, the error could read something more like this:

> Could not determine an app name from the given app_path "/". Please provide an app name with the --app-name option

Which would make much more sense to me — the problem isn't that we can't write to `"/"`, but it's that we can't work out an app name to use given the path provided. This kind of error message would better reflect what's going on, and would also be a useful way to help our users discover this extra option in the case where they're clearly trying to do something unusual.